apps/web/app/api/v1/auth.ts

```diff
@@ -0,0 +1,37 @@
+import { authOptions } from "@/app/api/auth/[...nextauth]/authOptions";
+import { getServerSession } from "next-auth";
+import { getApiKeyFromKey } from "@formbricks/lib/services/apiKey";
+import { getEnvironmentBySession } from "@formbricks/lib/services/environment";
+import { TAuthentication, TAuthenticationApiKey, TAuthenticationSession } from "@formbricks/types/v1/auth";
+
+export async function getAuthentication(request: Request): Promise<TAuthentication | null> {
+  const apiKey = request.headers.get("x-api-key");
+  const session = await getServerSession(authOptions);
+
+  if (apiKey) {
+    const apiKeyData = await getApiKeyFromKey(apiKey);
+    if (apiKeyData) {
+      const authentication: TAuthenticationApiKey = {
+        type: "apiKey",
+        environmentId: apiKeyData.environmentId,
+      };
+      return authentication;
+    }
+  } else if (session) {
+    const environmentData = await getEnvironmentBySession(session.user);
```

Comment:
As I understand this, this function returns the first environment the user belongs to. But later we need the environment the resource belongs to. So this isn't working.
I think here we are unable to provide a `environmentId` and we should just save the `userId` in the authentication object and later derive from this userId if the user has access. It shifts the complexity to the authorization check but I think it's the only option. What do you think?